Scripts/Action/ToggleAction.cs

```diff
@@ -6,21 +6,21 @@
     public class ToggleAction : BooleanAction
     {
         /// <inheritdoc />
-        public override void Receive(bool value, object sender = null)
+        public override void Receive(bool value, object initiator = null)
         {
             previousValue = Value;
             Value = value;
 
             if (!State)
             {
-                OnActivated(true);
+                OnActivated(true, initiator);

```

Comment:
again, should be `initiator ?? this` in case initiator is null then you want to use that script